### ISSUE TITLE
No matching devices found.

### ISSUE THREAD

**Issue by Entropy-Xu** [ITCMD/RunAsGpu#1]
When I am going to launch the RTX Voice with this programe,it shows "No matching devices found." and the RTX Voice shows "Unable to start microphone denoising"

**Comment by Entropy-Xu** [comment]
I have two GPU installed on my PC, GTX 1060 5GB and GTX 950

**Comment by ITCMD** [comment]
Hmm interesting. You've selected the GPU that is your primary one? Sounds like the program is failing to temporarily disable your primary gpu.

The rtx voice error should be unrelated as this program does not run the apps it starts in any special way.

**Comment by ITCMD** [comment]
Would you be able to show a recording of it?

**Comment by Entropy-Xu** [comment]
> Would you be able to show a recording of it?
https://youtu.be/hPwMtv2ygVs

**Comment by ITCMD** [comment]
Thanks for the video!
Seems like the program isn't prompting you for admin access for some reason. Could you right click the runas program and select "Run as administrator" to see if this is the issue?

**Comment by Entropy-Xu** [comment]
emmmm,i've tested it ,but the same problem still exist

**Comment by ITCMD** [comment]
Interesting!
I'm thinking it's probably grabbing the wrong device ID or... something. Not sure.

Is there a card.ini and cardname.ini file? Could you add the contents of those two files? Then, could you add what the actual device ID is for both of your graphics cards from Device manager? Screenshot shows how to get it:
![image](https://user-images.githubusercontent.com/32961763/170879831-59f846a8-ae84-4763-ac7b-115cd3453424.png)

**Comment by ITCMD** [comment]
Also, my version of the file I've been using is one day more recently edited than the version uploaded. Might be worth checking that one too (had to send in a zip to make github happy):
[runascard.zip](https://github.com/ITCMD/RunAsGpu/files/8793740/runascard.zip)

**Comment by Entropy-Xu** [comment]
emmm, it seems that the newest version still have the same problem

**Comment by SkeletoNCY** [comment]
![2](https://user-images.githubusercontent.com/53283266/171569450-5d5ae6d3-4475-4964-846b-c0820440a0f3.png)

**Comment by Entropy-Xu** [comment]
![3](https://user-images.githubusercontent.com/53283266/171569507-4b28d88f-143e-4d71-bd08-db2c8a6ce3db.png)
![1](https://user-images.githubusercontent.com/53283266/171569525-af425359-a894-462c-9985-afa8a5577fb8.png)
![2](https://user-images.githubusercontent.com/53283266/171569555-df1baeeb-f5d7-49bf-9e14-dbbcc667636e.png)

**Comment by Entropy-Xu** [comment]
emmm,i think the program grabbing the wrong device ID.

**Comment by ITCMD** [comment]
Yes it does! Thanks for those details! On my next day off I'll take a closer look at the code. Might have you run a few commands to see what they output perhaps.

In the meantime, you can also manually disable the gpu in device manager, launch rtx voice, then re-enable it to basically do what the program does. Clunky, but if you need it that will work until I have this fixed.

**Comment by Entropy-Xu** [comment]
Yep! I m manually disable my gpu right now,xD

**Comment by ITCMD** [comment]
Re-Opening just so I dont forget about this. I do plan to look into it more in the coming weeks to fix this if possible :P

Let me know if you want me to start a different issue to stop you from getting pinged with updates.

**Comment by iamafk** [comment]
having the same issue as OP, 3070ti & 1660S.

**Comment by AG-w** [comment]
I'm here to save the day
you don't need extra escape character when you export card.ini

open that batch file
in this section
```
for /f "tokens=1,2,3* delims=^&" %%A in ('echo "!card%cardnum%path!"') do (
	set tempvar=%%~A^^^&%%~B*
	echo %%~A^^^&%%~B*>card.ini
)
```
remove 2 "^" from echo
it should looks like this now
```
for /f "tokens=1,2,3* delims=^&" %%A in ('echo "!card%cardnum%path!"') do (
	set tempvar=%%~A^^^&%%~B*
	echo %%~A^&%%~B*>card.ini
)
```

now it can detect device correctly

**Comment by ITCMD** [comment]
Excellent find! Apologies for kind of forgetting about this... I havent had a non-F intel computer in a while, so Ive had no use for it nor the ability to test much.

Ill look to implement this.
